cdap-common/src/main/java/io/cdap/cdap/common/lang/DirectoryClassLoader.java

```diff
@@ -134,7 +135,7 @@ protected boolean needIntercept(String className) {
   }
 
   private static void addJarURLs(File dir, List<URL> result) throws MalformedURLException {
-    for (File file : DirUtils.listFiles(dir, "jar")) {
+    for (File file : new TreeSet<>(DirUtils.listFiles(dir, "jar"))) {
```

Comment:
What's the purpose of the natural/alphabetic ordering here?

Reply:
Primarily to ensure the top level artifact we put into root dir always go first (it's name starts with space that would bring it to the top of file list). Most file systems already return in proper order, but it's better be safe.
Also it would provide determinism in general. When a resource is available in multiple jars it will always go in same order.